packages/react/src/components/theme-switcher/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react';
+import Button from 'devextreme-react/button';
+import { currentTheme, refreshTheme } from 'devextreme/viz/themes';
+import './ThemeSwitcher.scss';
+
+type Theme = 'dark'| 'light';
+
+const storageKey = 'themeViewer';
+
+function getCurrentTheme(): Theme {
+  return window.localStorage[storageKey] || 'light';
+}
+
+function switchTheme(themeName?: Theme) {
+  const themePrefix = 'theme-';
+  const body = document.querySelector('body');
+
+  if (!body) {
+    return;
+  }
+
+  themeName = themeName || getCurrentTheme();
+
+  body.className = `${body.className.replace(/\s*theme-((light)|(dark))\s*/, ' ')} ${themePrefix + themeName}`;
+
+  window.localStorage[storageKey] = themeName;
+
+  currentTheme(`material.${themeName}`);
+  refreshTheme();
+}
+
+export const ThemeSwitcher = () => {
+  const [theme, setTheme] = useState(getCurrentTheme());
```

Comment:
It is better to move this functionality into the context on the top level of the app, ThemeSwitcher can get the current theme from the context and will call its switchTheme method.

Reply:
+1 we used similar techique in the aforementioned [example](https://github.com/DevExpress-Examples/devextreme-project-template-theme-selector/commit/f4c0a7a6a3635a49e17d19093cc6b8a5f4281cae#diff-21c51e1927ad1b17c953369ea14a6d9e3bd117395352fc72cba01949c2b47fa9)